sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Scenarios/BufferedProducerTest.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Producer;
+using Azure.Messaging.EventHubs.Tests;
+
+namespace Azure.Messaging.EventHubs.Stress
+{
+    internal class BufferedProducerTest
+    {
+        private Metrics metrics;
+        private BufferedProducerTestConfig testConfiguration;
+
+        public BufferedProducerTest(BufferedProducerTestConfig configuration)
+        {
+            testConfiguration = configuration;
+            metrics = new Metrics(configuration.InstrumentationKey);
+        }
+
+        public async Task Run()
+        {
+            var connectionString = testConfiguration.EventHubsConnectionString;
+            var eventHubName = testConfiguration.EventHub;
+            var appInsightsKey = testConfiguration.InstrumentationKey;
+            var durationInHours = testConfiguration.DurationInHours;
+
+            metrics = new Metrics(appInsightsKey);
+            using var enqueueingCancellationSource = new CancellationTokenSource();
+
+            var runDuration = TimeSpan.FromHours(durationInHours);
+            enqueueingCancellationSource.CancelAfter(runDuration);
+
+            var enqueuingTasks = default(IEnumerable<Task>);
+
+            try
+            {
+                enqueuingTasks = Enumerable
+                    .Range(0, 2)
```

Comment:
I wonder... do we need more than 2?   Generally, we advise customers to use a single instance in their application unless they have unusual needs for a second connection.    Having a second instance in this test is helpful to ensure that we don't have any shared state that would have issues when multiple clients are used.   But I don't know that we have value above 2.